core/vm/jump_table.go

```diff
@@ -78,6 +79,17 @@ func validate(jt JumpTable) JumpTable {
 	return jt
 }
 
+func newShanghaiInstructionSet() JumpTable {
+	instructionSet := newMergeInstructionSet()
+	instructionSet[SENDALL] = &operation{
+		execute:    opSendall,
+		dynamicGas: gasSelfdestructEIP3529,
```

Comment:
I think you forgot to charge constant gas too
```suggestion
		constantGas: params.SelfdestructGasEIP150,
		dynamicGas: gasSelfdestructEIP3529,
```